src/coreComponents/physicsSolvers/fluidFlow/StencilOutput.hpp

```diff
@@ -0,0 +1,169 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file StencilOutput.hpp
+ */
+
+#ifndef SRC_CORECOMPONENTS_PHYSICSSOLVERS_FLUIDFLOW_STENCILOUTPUT_HPP_
+#define SRC_CORECOMPONENTS_PHYSICSSOLVERS_FLUIDFLOW_STENCILOUTPUT_HPP_
+
+#include "physicsSolvers/FieldStatisticsBase.hpp"
+#include "physicsSolvers/fluidFlow/FlowSolverBase.hpp"
+
+namespace geos
+{
+
+/**
+ * @class StencilOutput
+ * Task class allowing the ouput of communicating elements data.
+ * Only current usage is to output the cell-cell transmissibility (= "CellToCellOutput"), but we could add more
+ * implementations to output the transmissibility of boundary or surface connections (or everything that is managed
+ * by the stencils).
+ */
+class StencilOutput : public FieldStatisticsBase< FlowSolverBase >
+{
+public:
+
+  /**
+   * @brief Constructor for the statistics class
+   * @param[in] name the name of the task coming from the xml
+   * @param[in] parent the parent group of the task
+   */
+  StencilOutput( const string & name,
+                 Group * const parent );
+
+  /// Accessor for the catalog name
+  static string catalogName() { return "CellToCellOutput"; }
+
+  /**
+   * @defgroup Tasks Interface Functions
+   *
+   * This function implements the interface defined by the abstract TaskBase class
+   */
+  /**@{*/
+
+  virtual bool execute( real64 const time_n,
+                        real64 const dt,
+                        integer const cycleNumber,
+                        integer const eventCounter,
+                        real64 const eventProgress,
+                        DomainPartition & domain ) override;
+
+  /**@}*/
+
+private:
+
+  using Base = FieldStatisticsBase< FlowSolverBase >;
+  using LocalToGlobalMap = ElementRegionManager::ElementViewAccessor< arrayView1d< globalIndex const > >;
+
+  class StencilOutputKernel;
+
+  /**
+   * @brief Element-element connection data extracted from the kernel.
+   */
+  struct KernelConnectionData
+  {
+    real64 transmissibility[2];
+    int regionId[2];
+    int subRegionId[2];
+    int elementId[2];
+  };
+
+  /**
+   * @brief Element-element connection data to be output by the class.
+   */
+  struct ConnectionData
+  {
+    real64 transmissibility[2];
+    int globalId[2];
+
+    /**
+     * @brief minus operator to be able to sort instances before outputing them.
+     * The sorting is done by the first global id, then by the second global id.
+     * @param other compared instance
+     * @return true if this instance is to sort before the other instance.
+     */
+    bool operator<( ConnectionData const & other ) const
+    {
+      return globalId[0] != other.globalId[0] ? globalId[0] < other.globalId[0] :
+             globalId[1] < other.globalId[1];
+    }
+  };
+
+
+  /**
+   * @brief the list of the current registered output target filepaths.
+   */
+  std::set< string > outputFiles;
```

Comment:
I was a bit confused by this one. Maybe add the `m_` prefix to make it easier to work out that it's a member variable.